chutney/ui/src/app/modules/campaign/components/execution/detail/campaign-execution.component.ts

```diff
@@ -1,180 +1,303 @@
+        }
+        return null;
+    }
+
+    sortBy(property) {
+        if (this.orderBy === property) {
+            this.reverseOrder = !this.reverseOrder;
+        }
+        this.orderBy = property;
+
+        return sortByAndOrder(
+            this.report.report.scenarioExecutionReports,
+            (i) => i[property] == null ? '' : i[property],
+            this.reverseOrder
+        );
+    }
+
+    exportReport() {
+        this.campaignService.findExecution(this.report.report.executionId)
+            .subscribe({
+                next: (report: CampaignExecutionFullReport) => {
+                    const url = location.origin + '/#' + this.router.serializeUrl(this.router.createUrlTree(['/scenario', ':id', 'executions']));
+                    const pdf = new jsPDF('l', 'mm', 'a4');
+                    const pdfFontSize = pdf.getFontSize();
+                    const duration = new DurationPipe();
+
+
+                    const docTitle = report.campaignName;
+                    pdf.text(docTitle, 148, 20, { align: 'center' });
+
+                    const docRecap = duration.transform(Number.parseInt(this.report.report.duration)) + ' - ' + this.report.passed + ' OK / ' + this.report.total;
+                    pdf.setFontSize(pdfFontSize - 4);
+                    pdf.text(docRecap, 148, 30, { align: 'center' });
+
+                    const dataHeader = [["id", "Scenario", "Status", "error"]];
+                    const dataBody = report.scenarioExecutionReports.map(s => [s.scenarioId.toString(), s.testCaseTitle, s.status, s.error.toString()]);
+                    pdf.setFontSize(pdfFontSize - 2);
+                    autoTable(pdf, {
+                        body: dataBody,
+                        head: dataHeader,
+                        startY: 40,
+                        theme: 'striped',
+                        useCss: true,
+                        didParseCell(data) {
+                            setStatusStyle(data);
+                        },
+                        didDrawCell(data) {
+                            if (data.cell.section === 'body' && data.column.index === 1) {
+                                let scenarioId = data.row.cells[0].raw.toString();
+                                pdf.link(data.cell.x, data.cell.y, data.cell.width, data.cell.height, { url: url.replace(':id', scenarioId) });
+                            }
+                        }
+                    });
+
+                    pdf.addPage();
+                    this.translate.get('campaigns.execution.scenarios.title').subscribe((res: string) => {
+                        pdf.text(res, 148, 15, { align: "center" });
+                    });
+
+                    const scenarioReportHeader = [["step", "Status", "error"]];
+
+                    report.scenarioExecutionReports
+                        .map(s => this.buildExecutionReport(s))
+                        .forEach(r => {

```

Comment:
if you add page here and not at line `224`, I think you can remove `pdf.deletePage(pdf.internal.pages.length - 1)` at line 249.